### PR TITLE
add a proof of concept on how to dynamically load plugin's debug cate…

### PR DESCRIPTION
**Pull request by mzanetti**
…gories

not ready for merging, just a proof of concept

### CODE REVIEW

File: libguh/devicemanager.cpp [hunk 1]
```diff
@@ -232,6 +232,35 @@ DeviceManager::~DeviceManager()
     }
 }
 
+QList<QJsonObject> DeviceManager::pluginNames()
```
Comment (mzanetti) [inline libguh/devicemanager.cpp]: This right now just copies parts of loadPlugins(), that's not good. Needs to share some code without calling "loader.instance()" in this static method.

File: libguh/devicemanager.h [hunk 2]
```diff
@@ -90,6 +91,8 @@ class DeviceManager : public QObject
     explicit DeviceManager(QObject *parent = 0);
     ~DeviceManager();
 
+    static QList<QJsonObject> pluginNames();
```
Comment (mzanetti) [inline libguh/devicemanager.h]: pluginNames is a bad name. should rather be something like pluginMetadata() or so

File: libguh/loggingcategories.cpp [hunk 3]
```diff
@@ -48,4 +48,4 @@ Q_LOGGING_CATEGORY(dcUdpCommander, "UdpCommander")
 Q_LOGGING_CATEGORY(dcWakeOnLan, "WakeOnLan")
 Q_LOGGING_CATEGORY(dcWemo, "Wemo")
 Q_LOGGING_CATEGORY(dcWifiDetector, "WifiDetector")
-Q_LOGGING_CATEGORY(dcKodi, "Kodi")
+//Q_LOGGING_CATEGORY(dcKodi, "Kodi")
```
Comment (mzanetti) [inline libguh/loggingcategories.cpp]: remove all plugin's categories from here

File: libguh/loggingcategories.h [hunk 4]
```diff
@@ -53,7 +53,7 @@ Q_DECLARE_LOGGING_CATEGORY(dcUdpCommander)
 Q_DECLARE_LOGGING_CATEGORY(dcWakeOnLan)
 Q_DECLARE_LOGGING_CATEGORY(dcWemo)
 Q_DECLARE_LOGGING_CATEGORY(dcWifiDetector)
-Q_DECLARE_LOGGING_CATEGORY(dcKodi)
+//Q_DECLARE_LOGGING_CATEGORY(dcKodi)
```
Comment (mzanetti) [inline libguh/loggingcategories.h]: and here

File: plugins/deviceplugins/kodi/devicepluginkodi.json [hunk 5]
```diff
@@ -1,5 +1,6 @@
 {
     "name": "Kodi",
+    "idName": "Kodi",
```
Comment (mzanetti) [inline plugins/deviceplugins/kodi/devicepluginkodi.json]: add an idName to all plugins. The resulting category will be "dc" + idName

File: plugins/deviceplugins/kodi/jsonhandler.cpp [hunk 6]
```diff
@@ -19,7 +19,7 @@
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
 #include "jsonhandler.h"
-#include "loggingcategories.h"
+#include "extern-plugininfo.h"
```
Comment (mzanetti) [inline plugins/deviceplugins/kodi/jsonhandler.cpp]: include this in plugin's files, except the main .cpp of a plugin. use the non-extern one there.

File: plugins/generateplugininfo.py [hunk 7]
```diff
@@ -117,12 +121,37 @@ def extractEventTypes(deviceClassMap):
...
+out2(" */")
+out2("#ifndef PLUGININFO_H")
+out2("#define PLUGININFO_H")
+out2("#include \"typeutils.h\"")
+out2("#include <QLoggingCategory>")
+out2("")
+if 'idName' in pluginMap:
+    out2("Q_DECLARE_LOGGING_CATEGORY(dc%s)" % pluginMap['idName'])
+out2("")
+out2("#endif")
```
Comment (mzanetti) [inline plugins/generateplugininfo.py]: this now generates a second plugininfo.h, but only declaring extern stuff. This also would solve the issue when you need the uuids in some other places than the plugin's main .cpp. you can call extractVendors() in here too, but prefix things with "extern" somehow.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/278/
Test PASSed.

**Pull request closed without merging**